internal/api/mfa.go

```diff
@@ -245,7 +246,45 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 		return badRequestError("%v has expired, verify against another challenge or create a new challenge.", challenge.ID)
 	}
 
-	if valid := totp.Validate(params.Code, factor.Secret); !valid {
+	valid := totp.Validate(params.Code, factor.Secret)
+	if config.Hook.MFA.IsEnabled() {
+		// To allow for future cases where we don't know that the payload is going to be passed in
+
+		payload, err := CreateMFAVerificationHookInput(user.ID, factor.ID, valid)
+		if err != nil {
+			return err
+		}
+
+		h := AuthHook{
+			event:    MFAVerificationEvent,
+			payload:  payload,
+			hookType: PostgresHook,
+			// TODO: find a better way to relay this
+			db: a.db,
+		}
+
+		resp, err := h.Trigger()
+		if err != nil {
+			return err
+		}
+		parsedErrorResponse, err := parseErrorResponse(resp)
+		if err != nil {
+			return err
+		}
+		if parsedErrorResponse != nil {
+			return errors.New(parsedErrorResponse.ErrorMessage)
+		}
+		// TODO: Decide what to do here
+		response, err := parseMFAVerificationResponse(resp)
+		if err != nil {
+			return err
+		}
```

Comment:
In general I would prefer that errors and retries are handled within `Trigger`, so this whole section either deals with a `AuthHookError` or something or a successful response. I'd also prefer if we could get the typed response out as well. No need to parse it here again; it's an implementation detail from this section's point of view.